src/lib/api/ext/class.ts

```diff
@@ -154,51 +124,50 @@ export class ClassDirective extends BaseFxDirective
   // ******************************************************************
 
   /**
-   * Build an mqActivation object that bridges
-   * mql change events to onMediaQueryChange handlers
+   * Configure adapters (that delegate to an internal ngClass instance) if responsive
+   * keys have been defined.
    */
-  protected _configureMQListener() {
-    const value = this._classAdapter.queryInput('class');
-    this._classAdapter.listenForMediaQueryChanges('class', value, (changes: MediaChange) => {
-      this._updateKlass(changes.value);
-    });
-
-    this._ngClassAdapter.listenForMediaQueryChanges('ngClass', value, (changes: MediaChange) => {
-      this._updateNgClass(changes.value);
-      this._ngClassInstance.ngDoCheck();    // trigger NgClass::_applyIterableChanges()
-    });
+  protected _configureAdapters() {
+    this._base = new BaseFxDirectiveAdapter(
+        'ngClass', this.monitor, this._ngEl, this._renderer
+    );
+    if (!this._ngClassInstance) {
+      // Create an instance NgClass Directive instance only if `ngClass=""` has NOT been defined on
+      // the same host element; since the responsive variations may be defined...
+      let adapter = new RendererAdapter(this._renderer);
+      this._ngClassInstance = new NgClass(
+          this._iterableDiffers, this._keyValueDiffers, this._ngEl, <any> adapter
+      );
+    }
+    this._fallbackToKlass();
   }
 
   /**
-   *  Apply updates directly to the NgClass:klass property
-   *  ::ngDoCheck() is not needed
+   * Build an mqActivation object that bridges mql change events to onMediaQueryChange handlers
+   * NOTE: We delegate subsequent activity to the NgClass logic
+   *       Identify the activated input value and update the ngClass iterables...
+   *       Use ngDoCheck() to actually apply the values to the element
    */
-  protected _updateKlass(value?: NgClassType) {
-    let klass = value || this._classAdapter.queryInput('class');
-    if (this._classAdapter.mqActivation) {
-      klass = this._classAdapter.mqActivation.activatedInput;
-    }
-    this._ngClassInstance.klass = klass;
+  protected _configureMQListener(baseKey = 'ngClass') {
+    const fallbackValue = this._base.queryInput(baseKey);
+    this._base.listenForMediaQueryChanges(baseKey, fallbackValue, (changes: MediaChange) => {
+      this._ngClassInstance.ngClass = changes.value || '';
+      this._ngClassInstance.ngDoCheck();
+    });
   }
 
   /**
-   *  Identify the activated input value and update the ngClass iterables...
-   *  needs ngDoCheck() to actually apply the values to the element
+   * Initial lookup of raw 'class' value (if any)
    */
-  protected _updateNgClass(value?: NgClassType) {
-    if (this._ngClassAdapter.mqActivation) {
-      value = this._ngClassAdapter.mqActivation.activatedInput;
+  protected _fallbackToKlass() {
+    if (!this._base.queryInput('ngClass')) {
+      this.ngClassBase = this._getAttributeValue('class');
```

Comment:
Premise: When a breakpoint deactivates, the fallback value in the non-responsive input will be used.
But what happens when the fallback/default input is not defined?

Consider this logic here in `_fallbackToKlass()`. Is this logic valid?

This methods effectively says that if a responsive breakpoint deactivates (e.g. `[ngClass.md]`) and the default `[ngClass]` has not been defined, then we should use the `class` attribute value (if any) as the fallback value.

The challenge is the any `ngClass` responsive input could be a string, array, or map. Each do different things.

1) Is this a valid assumption to fallback/restore the initial `class` value?